stingray/pulse/overlapandsave/ols.py

```diff
@@ -0,0 +1,355 @@
+"""
+The code in this file was copied from
+https://github.com/fasiha/array_range
+https://github.com/fasiha/nextprod-py
+https://github.com/fasiha/overlap_save-py
+
+Thanks to Ahmed Fasih for releasing this to the larger public with the
+Unlicense.
+
+# Fast-convolution via overlap-save: a partial drop-in replacement for scipy.signal.fftconvolve
+
+Features:
+
+- 1D and 2D (both tested) and higher (untested) arrays
+  - (Currently unsupported is convolving different-dimensional signals)
+- Real and complex arrays
+- Memory-mapped input/outputs fully supported (and tested)
+- Supports alternative FFT engines such as PyFFTW
+- Supports reflect-mode (signal assumed to reflect infinitely, instead of 0
+  outside its support; useful for avoiding edge effects)
+- Relatively straightforward to paralellize each step of the algorithm
+- Extensively unit-tested
+
+When it can be used as a drop-in replacement for `fftconvolve`:
+
+- when you call `fftconvolve` with `mode='same'` and `axes=None`
+- [See docs](https://docs.scipy.org/doc/scipy/reference/generated/scipy.signal.fftconvolve.html)
+
+Example (culled from unit tests):
+```py
+import numpy as np
+from scipy.signal import fftconvolve
+
+# Generate a 100x100 signal array and a 10x10 filter array
+nx = 100
+nh = 10
+x = np.random.randint(-10, 10, size=(nx, nx)) + 1.0
+h = np.random.randint(-10, 10, size=(nh, nh)) + 1.0
+
+# Compute the linear convolution using the FFT, keeping the center 100x100 samples
+expected = fftconvolve(x, h, mode='same')
+
+# Use overlap-save, computing the output in 6x5-sized chunks. Instead of one huge FFT, we do a
+# several tiny ones
+from ols import ols
+actual = ols(x, h, [6, 5])
+
+# The two will match
+assert np.allclose(expected, actual)
+```
+"""
+
+import numpy as np
+
+from itertools import product
+from typing import List, Iterable, Tuple
+
+from typing import List
+import math
```

Comment:
Is there a reason to use `math` here rather than `numpy`?

Reply:
See below